src/components/element-editor/validation/validation.tsx

```diff
@@ -0,0 +1,103 @@
+import { useState } from "react";
+import { MultiEntry } from "src/components/common";
+import { ValidationChild } from "./validation-child";
+import { EElementType, useBuilder } from "src/context-providers";
+import { ELEMENT_VALIDATION_TYPES } from "src/data";
+import { SimpleIdGenerator } from "src/util/simple-id-generator";
+
+export interface IEntryProps {
+    type?: string;
+    rule?: string;
+    errorMessage?: string;
+}
+
+export const Validation = () => {
+    // =========================================================================
+    // CONST, STATES, REFS
+    // =========================================================================
+    const { focusedElement } = useBuilder();
+    const element = focusedElement.element;
+    const [numberOfChildEntry, setNumberOfChildEntry] = useState(0);
+    const [childEntryValues, setChildEntryValues] = useState<IEntryProps[]>([]);
+    const [childKey, setChildKey] = useState([]);
```

Comment:
proposing to use `childEntryValues` to track the no. of entries so you can remove  `childKey` and `numberOfChildEntry`.

I see that `childKey` is used for the `key` prop, you can use something like `validation-entry-${index}` in place of it